### PR TITLE
implemented random access paging for query then fetch

### PR DESCRIPTION
**Pull request by mfelsche**
without caching. every page request will hit the shards

### CODE REVIEW

File: core/src/main/java/io/crate/Constants.java [hunk 1]
```diff
@@ -50,4 +51,6 @@
     public static final Set<String> INVALID_TABLE_NAME_CHARACTERS = ImmutableSet.of(".");
 
     public static final Predicate<CharSequence> INVALID_COLUMN_NAME_PREDICATE = Predicates.contains(Pattern.compile("[\\[\\'\\]]"));
+
+    public static final String LF = OsUtils.WINDOWS ? "\r\n" : "\n";
```
Comment (mfussenegger) [inline core/src/main/java/io/crate/Constants.java]: what's wrong with System.lineSeparator() ?

File: sql/src/test/java/io/crate/executor/transport/PagingTasksTest.java [hunk 2]
```diff
@@ -681,4 +677,347 @@ public void testPagedNestedLoopWithProjectionsBothSidesPageable() throws Excepti
...
+        closeMeWhenDone = gappedResult;
+        assertThat(TestingHelpers.printedPage(gappedResult.page()),
+                is("5| Matthias Wahl| false\n" +
+                   "6| Philipp Bogensberger| false\n")
+        );
+
+    }
+
+    @Test
+    public void queryThenFetchPageWithBigGap() throws Exception {
```
Comment (dobe) [inline sql/src/test/java/io/crate/executor/transport/PagingTasksTest.java]: prefix test methods with test*

File: sql/src/test/java/io/crate/testing/TestingHelpers.java [hunk 3]
```diff
@@ -346,6 +346,22 @@ public void describeTo(Description description) {
         return column;
     }
 
+    /**
+     * Get the values at column index <code>index</code> within all <code>rows</code>
+     */
+    public static @Nullable Object[] getColumn(Page page, int index) throws Exception {
```
Comment (dobe) [inline sql/src/test/java/io/crate/testing/TestingHelpers.java]: where is this used?

**Comment by dobe** [comment]
ack

**Pull request merged**